### DISCUSSION TITLE
Save Metadata In A File With The Same Name

### DISCUSSION THREAD

**Discussion by Caterpillar-cmd** [ImageMagick/ImageMagick#3233] [Help] [answered]
Hello!

I am having some troubles with this:
` magick identify - verbose C:\Users\Home\Pic\image1.png > %t.txt`

Above is the command line in CMD windows. Could someone explain me why the "%t" do not return a file with the name "image1"? 

Best regards,
Cat-cmd

**Comment by snibgo** [comment]
There is no reason why it should. The windows command interpreter runs your magick command, and sends any output from the magick stdout to the file called %t.txt.

Within the IM command, if you had a format that included %t, the %t would expand to the filename.

**Reply by Caterpillar-cmd** [reply]
Is it possible to have a format the includes the filename? I am a beginner with IM.

`magick identify - verbose C:\Users\Home\Pic\image1.png  >   C:\users\Home\others\%t.txt`

**Comment by snibgo** [comment]
%t is a feature of ImageMagick. You can't use it _outside_ ImageMagick. Outside IM, you need to give the full fulename:
```
magick identify -verbose C:\Users\Home\Pic\image1.png > C:\users\Home\others\image1.txt
```

**Reply by Caterpillar-cmd** [reply]
I understood, therefore after the '>' I am already out from IM. I will try find a way to save from inside, probably there is. 
Thank you Snibgo for the help.

**Comment by snibgo** [comment]
There is no need to use redirection ">", so:
```
magick toes.png -set filename:MYFILE %t -verbose info:%[filename:MYFILE].txt
```

**Reply by Caterpillar-cmd** [reply]
I tested first that way `magick C:\Users\Home\Pic\image1.png -set filename:MYFILE%t  -verbose info: %[filename:MYFILE].txt`
also : `magick C:\Users\Home\Pic\image1.png -set filename:MYFILE%t   identify -verbose info: %[filename:MYFILE].txt`

Itested some different ways, nevertheless some times nothing is shown in the prompt, some times was save a file named as %[filename without extension...

**Comment by snibgo** [comment, marked as answer]
You have removed the space between MYFILE and %t.

You have inserted a space after "info:"

Also, if you have this in a BAT script, you need to double the percentage signs: %%.

**Reply by Caterpillar-cmd** [reply]
Thank so much Snibgo, now is correct and I understood a lot of things!
